src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Controllers/DelegationsController.cs

```diff
@@ -80,6 +81,119 @@ public async Task<ActionResult> Post([FromBody] List<Rule> rules)
             }
         }
 
+        /// <summary>
+        /// Endpoint for deleting delegated rules between parties
+        /// </summary>
+        /// <response code="201" cref="List{Rule}">Deleted</response>
+        /// <response code="206" cref="List{Rule}">Partial Content</response>
+        /// <response code="400">Bad Request</response>
+        /// <response code="500">Internal Server Error</response>
+        [HttpPost]
+        [Authorize(Policy = AuthzConstants.ALTINNII_AUTHORIZATION)]
+        [Route("authorization/api/v1/[controller]/DeleteRules")]
+        public async Task<ActionResult> DeleteRule([FromBody] List<RequestToDelete> rulesToDelete)
+        {
+            if (rulesToDelete == null || rulesToDelete.Count < 1)
+            {
+                return BadRequest("Missing rulesToDelete in body");
+            }
+
+            if (rulesToDelete.Any(r => r.RuleIds == null || r.RuleIds.Count == 0))
+            {
+                return BadRequest("Not all RequestToDelete has RuleId");
+            }
+
+            if (rulesToDelete.GroupBy(x => PolicyHelper.GetAltinnAppDelegationPolicyPath(x.PolicyMatch)).Any(g => g.Count() > 1))
+            {
+                return BadRequest("Input should not contain any duplicate policies");
+            }
+
+            if (!ModelState.IsValid)
+            {
+                return BadRequest("Invalid model");
+            }
+
+            try
+            {
+                List<Rule> deletionResults = await _pap.TryDeleteDelegationPolicyRules(rulesToDelete);
+                int ruleCountToDelete = DelegationHelper.GetRulesCountToDeleteFromRequestToDelete(rulesToDelete);
+
+                if (deletionResults.Count == ruleCountToDelete)
+                {
+                    _logger.LogInformation("Deletion completed");
+                    return Created("Created", deletionResults);
+                }
+
+                if (deletionResults.Count > 0)
+                {
+                    _logger.LogInformation("Partial deletion completed");
+                    return StatusCode(206, deletionResults);
```

Comment:
The 206 here is the same as the 206 in the existing Add rule call:
public async Task<ActionResult> Post([FromBody] List<Rule> rules)

It might be good to differentiate when the caller has to investigate what went Ok and what has to be marked as a failure. If Ok no such test has to be performed. I'm open for not using 206 but would like to not have to use the same as when everything is Ok and 206 looked like a good alternative.

Reply:
We can add information about partial success inside the response model, or do we think the caller would use the status code to skip deserialization and ignore the response?

Reply:
On thin ice here but I se no reason to deserialize the model if everything went Ok then all the input is handled.

Reply:
And the handling is done in the same way as the Add rule also returning a 206 for partial success.

Reply:
Think that handling of the response is not necessary if 200 Ok but must be done with 206. as some of the changes expected to happen has not been performed. Those changes should be marked as failed.